experiment/nursery/main.go

```diff
@@ -46,26 +48,58 @@ var dir = flag.String("dir", "", "location to store cache entries on disk")
 var host = flag.String("host", "", "host address to listen on")
 var port = flag.Int("port", 8080, "port to listen on")
 
+// eviction knobs
+var minPercentBlocksFree = flag.Float64("min-percent-blocks-free", 10,
+	"minimum percent of blocks free on --dir's disk before evicting entries")
+var minPercentFilesFree = flag.Float64("min-percent-files-free", 10,
+	"minimum percent of blocks free on --dir's disk before evicting entries")
+var diskCheckInterval = flag.Duration("disk-check-interval", time.Minute,
+	"interval between checking disk usage (and potentially evicting entries)")
+
+// NOTE: remount is a bit of a hack, unfortunately the kubernetes volumes
+// don't really support this and to cleanly track entry access times we
+// want to use a volume with lazyatime (and not noatime or relatime)
+// so that file access times *are* recorded but are lazily flushed to the disk
+var remount = flag.Bool("remount", false,
+	"attempt to remount --dir with strictatime,lazyatime to improve eviction")
+
 func init() {
-	log.SetFormatter(&log.TextFormatter{})
+	log.SetFormatter(&log.JSONFormatter{})
 	log.SetOutput(os.Stdout)
 }
 
 func main() {
-	// TODO(bentheelder): bound cache size / convert to LRU
-	// TODO(bentheelder): improve logging
+	// TODO(bentheelder): add metrics
 	flag.Parse()
+	logger := log.WithFields(log.Fields{
```

Comment:
You should pass this logger to functions like `cacheHandler` and `monitorDiskAndEvict` instead of using the default undecorated logger or adding the field `component=nusery` in multiple places.

Reply:
I've been thinking about creating a custom formatter with the default field instead, there are lots of places the subpackages should log things that won't be decorated. We might want to do this in prow as well.